### DISCUSSION TITLE
Problem retrieving coordinate of a rectangle using CTkCanvas

### DISCUSSION THREAD

**Discussion by rzfNotahina** [TomSchimansky/CustomTkinter#2410] [Q&A] [answered]
Hi everyone, first time using customtkinter (also first time using gitHub). When I first saw it, I definitely likes it over the tkinter (even though, I still have to use it for some specific projects). Here's my problem: creating a rectangle on CTkCanvas is no problem, I get its ID, but when comes to retrieving the coords of the same rectangle, I get an error. Here's the code:
from customtkinter import CTk,CTkFrame,CTkCanvas

root = CTk()
root.geometry("800x400")

frame = CTkFrame(root)
frame.pack(fill="both", expand=True)

canvas = CTkCanvas(frame,width=50,height=50)
rect_id = canvas.create_rectangle(0, 0, 10, 10, fill = "gray")

x0, y0, x1, y1 = canvas.coords(rect_id)
print(x0, y0, x1, y1)

root.mainloop()

and here's the error I get: cannot unpack non-iterable NoneType object. I'm pretty sure it comes from the line where I try to retrieve the coords of the rectangle.
If anyone has any idea how to get those coords (not for rectangle only but for any shape in general) I'd like to hear about it. Thanks in advance

**Comment by rigvedmaanas** [comment, marked as answer]
Hi @rzfNotahina 

`CTkCanvas` is not meant to be used as a widget. It's used for internal widget rendering.
TomSchimansky mentions this in the issue #2066 
> CTkCanvas is not meant to be used outside the library itself, its just written for rendering the CTk components. 

So you should use `Canvas` directly from tkinter and It should work as intented.

Hope this helps

**Reply by rzfNotahina** [reply]
Thank you very much for your answer. I'll change my code for Canvas right away. Have a nice day.
